operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -236,38 +230,27 @@ func (r *RedpandaReconciler) reconcileDefluxed(ctx context.Context, rp *v1alpha2
 	log := ctrl.LoggerFrom(ctx)
 	log.WithName("RedpandaReconciler.reconcileDefluxed")
 
-	if !ptr.Deref(rp.Spec.ChartRef.UseFlux, true) {
-		// TODO (Rafal) Implement Redpanda helm chart templating with Redpanda Status Report
-		// In the Redpanda.Status there will be only Conditions and Failures that would be used.
-
-		if !atLeast(rp.Spec.ChartRef.ChartVersion) {
-			log.Error(fmt.Errorf("chart version needs to be at least %s", HelmChartConstraint), "", "chart version", rp.Spec.ChartRef.ChartVersion)
-			v1alpha2.RedpandaNotReady(rp, "ChartRefUnsupported", fmt.Sprintf("chart version needs to be at least %s. Currently it is %s", HelmChartConstraint, rp.Spec.ChartRef.ChartVersion))
-			r.EventRecorder.Eventf(rp, "Warning", v1alpha2.EventSeverityError, fmt.Sprintf("chart version needs to be at least %s. Currently it is %s", HelmChartConstraint, rp.Spec.ChartRef.ChartVersion))
-			// Do not error out to not requeue. User needs to first migrate helm release to at least 5.9.3 version
-			return nil
-		}
+	if ptr.Deref(rp.Spec.ChartRef.UseFlux, true) {
+		log.Info("useFlux is true; skipping non-flux reconciliation...")
+		return nil
 	}
-	return nil
-}
 
-func atLeast(version string) bool {
-	if version == "" {
-		return true
-	}
+	chartVersion := rp.Spec.ChartRef.ChartVersion
+	desiredChartVersion := redpanda.Chart.Metadata().Version
 
-	c, err := semver.NewConstraint(fmt.Sprintf(">= %s", HelmChartConstraint))
-	if err != nil {
-		// Handle constraint not being parsable.
-		return false
-	}
+	if !(chartVersion == "" || chartVersion == desiredChartVersion) {
+		msg := fmt.Sprintf(".spec.chartRef.chartVersion version needs to be %q or %q. got %q", desiredChartVersion, "", chartVersion)
 
-	v, err := semver.NewVersion(version)
-	if err != nil {
-		return false
+		log.Error(fmt.Errorf(msg), "", "chart version", rp.Spec.ChartRef.ChartVersion)
```

Comment:
Why there is empty string?

Should this be change to:
```suggestion
		log.Error(fmt.Errorf(msg), "chart version", rp.Spec.ChartRef.ChartVersion)
```

Reply:
That's been fixed! There's another use of an empty string but that's intentional so we log out `chart version must be "someversion" or ""`.